Add demo case pillow and topic

gherceg opened the pull request:
## Product Description
<!-- Where applicable, describe user-facing effects and include screenshots. -->

## Technical Summary
<!--
    Provide a link to any tickets, design documents, and/or technical specifications
    associated with this change. Describe the rationale and design decisions.
-->
https://dimagi.atlassian.net/browse/SAAS-18584

Adds separate pillow for demos to ensure minimal lag while we continue to improve the standard case pillow/topic to better handle load/reduce lag.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->
`DEMO_CASE_PILLOW`

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Tested on staging

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
No

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

gherceg committed:
Add demo case pillow and topic

gherceg committed:
Send changes to demo case topic when appropriate

This does not address the PillowError retry queue, which will still send
failed changes to the regular queue based on the doc type. Since the
purpose of the demo topic is to have the change quickly reflected in
the UI, it doesn't seem worthwhile to handle errors as well.

millerdev reviewed (approved):
:vomiting_face:

gherceg committed:
Add definition for demo case pillow in settings.py

gherceg committed:
Add demo case pillow config to expected test results

millerdev commented on corehq/pillows/case.py:
```diff
@@ -78,6 +87,7 @@ def get_case_pillow(
     processor_chunk_size=DEFAULT_PROCESSOR_CHUNK_SIZE,
     topics=None,
     dedicated_migration_process=False,
+    include_demo_topic=False,
```
Where is this used?

gherceg replied:
Decided to move it to a setting instead: 10b64fdfdd4fe5f3980800a23c0c6245c22745cb

millerdev reviewed (approved):
One non-blocking comment, although if `include_demo_topic` is unused maybe it should be removed?

gherceg committed:
Replace pillow parameter with setting

Given this only applies to the case pillow, it isn't worth updating the
run_ptop command to accept a new var to represent whether the case
pillow should read from both the case-sql and demo-case-sql topics.

millerdev commented on corehq/pillows/case.py:
```diff
@@ -102,10 +101,12 @@ def get_case_pillow(
             topics = CASE_TOPICS
+        else:
+            # drain changes from the demo case sql topic
+            topics = CASE_TOPICS + (DEMO_CASE_SQL,)
```
Didn't catch this before, but will this require that `DEMO_CASE_SQL` must exist on all environments? It seems ideal to only add it where it is needed, and not to have to worry about the additional complexity anywhere else.

Secondly, is it important that both topics have the same number of partitions?

gherceg committed:
Rely on feature flag for draining entirely

millerdev commented on corehq/pillows/case.py:
```diff
@@ -98,15 +98,8 @@ def get_case_pillow(
-        else:
-            # drain changes from the demo case sql topic
-            topics = CASE_TOPICS + (DEMO_CASE_SQL,)
+        assert set(topics).issubset(CASE_TOPICS), "This is a pillow to process cases only"
```
Will this fail if `topics == [DEMO_CASE_SQL]`?

gherceg replied:
Yeah it will, sorry rushed that commit.

gherceg replied:
933374001303dc4b7c0ade7d7ae5f0740596f4b3

gherceg committed:
Fix assertion to include demo case topic

gherceg committed:
Pass in all kwargs

gherceg committed:
Add feature flag for using demo case pillow
